src/proof/proof_node_manager.cpp

```diff
@@ -164,13 +165,10 @@ std::shared_ptr<ProofNode> ProofNodeManager::mkScope(
         for (const Node& acc : ac)
         {
           Node accr = d_rewriter->rewrite(acc);
-          if (accr != acc)
```

Comment:
you could turn this into an assertion then?

Reply:
No, the old code was deficient: we should map the assertion to itself if it does not rewrite (this is the 2nd point in my comment on the PR).

Reply:
Ah, I see it now.